src/app.ts

```diff
@@ -0,0 +1,111 @@
+/* eslint-disable camelcase */
+import { Bytes, KeyType, PrivateKey, PublicKey } from '@greymass/eosio';
+import randomBytes from 'randombytes';
+import { ES256KSigner, createJWT } from 'did-jwt';
+import * as jose from 'jose';
+import { PublicKey as PublicKeyCon } from 'eosjs/dist/eosjs-key-conversions';
+function generateRandomKeyPair(): { privateKey: PrivateKey; publicKey: PublicKey } {
+    const bytes = randomBytes(32);
+    const privateKey = new PrivateKey(KeyType.K1, new Bytes(bytes));
+    const publicKey = privateKey.toPublic();
+    return { privateKey, publicKey };
+}
+
+async function onPressLogin(redirect: string): Promise<string> {
+    const { privateKey, publicKey } = generateRandomKeyPair();
+    const payload = {
+        number: Math.floor(Math.random() * 100),
+        // domain: window.location.hostname,
+        redirect,
+        pubkey: publicKey.toString(),
+    };
+
+    const signer = ES256KSigner(privateKey.data.array, true);
+
+    const jwk = await createJWK(publicKey);
+    const issuer = toDid(jwk);
+    const token = await createJWT(payload, { issuer, signer, alg: 'ES256K-R' });
+    return token;
+}
+
+const createJWK = (publicKey: PublicKey) => {
```

Comment:
lets move `createJWK`, `toDid`, `toDidDocument` and `resolve` to a new file in util/did-jwk.ts as it hasn't got anything directly to do with app's in Tonomy

Reply:
also put a link to the source you copied from for our reference in `did-jwk.ts`